crates/store/src/filesystem/utility.rs

```diff
@@ -0,0 +1,42 @@
+// SPDX-FileCopyrightText: 2022 Profian Inc. <opensource@profian.com>
+// SPDX-License-Identifier: AGPL-3.0-only
+
+use crate::Error;
+
+use cap_async_std::fs::DirEntry;
+use serde::Serialize;
+use sha2::{Digest, Sha256};
+
+pub const ALGORITHM_NAME: &str = "sha256";
+
+pub fn sha256(buffer: &[u8]) -> String {
```

Comment:
We already have a module and functionality for this, at the very least this utility should utilize the existing functionality and reside within that module.
I also think we should use a consistent, standard format used in that module for simplicity of inspecting the store and debugging issues

Reply:
I can't use `sha-256=:TltwxsQjHl7DcPPZqQ/ZPMbZNtSdo2ms9am1EBBnSpU=:` for a filename so I guess I'll try and find some other way of representing it if thats available. I might have to add something to do that.

Reply:
Right now I feel like I'm going to have to substitute characters in the base64 so I can use it or just take the bytes and write them as lowercase like I did before. Seems like a more complicated interface to do the same thing but maybe I'm using it wrong here.

```rust
async fn sha256(buffer: &[u8]) -> Result<String, Error> {
    let algo = Algorithm::Sha256;
    let mut set = BTreeSet::new();
    set.insert(algo);
    let mut buf: Vec<u8> = Vec::with_capacity(buffer.len());
    let mut hasher = Algorithms::from(set).writer(&mut buf);
    hasher.write_all(buffer).await.map_err(Error::Io)?;
    Ok(hasher.digests().get(&algo).unwrap().to_string())
}
```